client/utils/date-time.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { dateI18n } from '@wordpress/date';
+
+type DateTimeFormat = string | null;
+
+interface FormatDateTimeOptions {
+	includeTime?: boolean; // Whether to include time in the formatted string (defaults to true)
+	useGmt?: boolean; // Whether to display the time in GMT/UTC (defaults to false)
+	separator?: string; // Separator between date and time (defaults to ' / ')
+	customFormat?: DateTimeFormat; // Custom format to use instead of WordPress settings
+}
+
+/**
+ * Formats a date and time string according to WordPress settings or a custom format.
+ *
+ * @param { string | Date } dateTime - The date and time string (e.g., '2024-10-23 15:28:26') or a JS Date object.
+ * @param { FormatDateTimeOptions } options - Additional options to control time inclusion and whether to use GMT/UTC.
+ * @return { string } - The formatted date and time string.
+ */
+export function formatUserDateTime(
+	dateTime: string | Date,
+	options: FormatDateTimeOptions = {
+		includeTime: true,
```

Comment:
I feel the `false` scenario for `includeTime` is the more common one, could we update the default value to reflect that? 🤔

Reply:
Yes, good point, addressed here [70b3c92](https://github.com/Automattic/woocommerce-payments/pull/9635/commits/70b3c921a7088370a6c7e1f503489a1b3902cb89) thanks!